docs/advanced_topics_performance_tuning.rst

```diff
@@ -101,15 +101,24 @@ Direct usage of ``cudaMallocHost``, while discouraged, can be forced by specifyi
 ``DALI_USE_PINNED_MEM_POOL=0`` in the environment.
 
 For device memory, DALI uses a stream-aware memory pool built on top of CUDA VMM resource
-(if the platform supports VMM) or plain ``cudaMalloc`` otherwise. The pool can be disabled by
-setting ``DALI_USE_DEVICE_MEM_POOL=0``. Use ``DALI_USE_VMM=0`` to switch from CUDA VMM to
+(if the platform supports VMM). It can be changed to ``cudaMallocAsync`` or even plain
 ``cudaMalloc``.
+In order to skip memory pool entirely and use ``cudaMalloc`` (not recommended), set
+``DALI_USE_DEVICE_MEM_POOL=0``.
+Set ``DALI_USE_CUDA_MALLOC_ASYNC=1`` to use ``cudaMallocAsync`` instead of DALI's internal memory
+pool.
+When using the memory pool (``DALI_USE_DEVICE_MEM_POOL=1`` or unset), you can disable the use of
+VMM by setting ``DALI_USE_VMM=0``. This will cause ``cudaMalloc`` to be used as an upstream memory
+resource for the internal memory pool.
+
+Using ``cudaMallocAsync`` results in slightly slower execution, but it enables memory pool sharing
```

Comment:
```suggestion
Using ``cudaMallocAsync`` results in execution time dependent on the CUDA toolkit and driver version, but it enables memory pool sharing
```
I don't think we can for sure claim that is slower in all cases.

Reply:
I can add some weasel word like "typically" and call it a day.